src/map-editor/components-smart/PopupCreateTechnique.tsx

```diff
@@ -64,14 +65,22 @@ export default class extends Component<Props, State> {
             throw new Error();
         }
 
+        if (techniqueData.when === undefined) {
+            PopupsContainer.alertPopup("Error", "Style is missing mandatory when condition.");
+            return;
+        }
+
         const style = {
             technique: techniqueData.technique,
-            when: techniqueData.when,
+            when:
+                typeof techniqueData.when === "string"
+                    ? Expr.parse(techniqueData.when).toJSON()
+                    : techniqueData.when,
```

Comment:
Do we really need to parse the expression?
after auto-formatting, they look scary.

Reply:
yes, we have to because the old syntax doesn't support many the many features that are needed by some of the data sources we're using (e.g. accessing properties containing `:` and `.` in their name). Also, we want to drop the old syntax at some point.
I agree that the indentation is terrible but that can be fixed by using prettier or own jsonc-parser based formatter.